identity/migration/src/main/java/io/camunda/identity/migration/AuthorizationMigrationHandler.java

```diff
@@ -0,0 +1,134 @@
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorizationMigrationHandler {
+
+  private final AuthorizationServices<AuthorizationRecord> authorizationService;
+  private final ManagementIdentityProxy managementIdentityProxy;
+
+  public AuthorizationMigrationHandler(
+      final AuthorizationServices<AuthorizationRecord> authorizationService,
+      final ManagementIdentityProxy managementIdentityProxy) {
+    this.authorizationService = authorizationService;
+    this.managementIdentityProxy = managementIdentityProxy;
+  }
+
+  public void migrate() {
+    UserResourceAuthorization lastRecord = null;
+    while (true) {
+      final List<UserResourceAuthorization> authorizations =
+          managementIdentityProxy.fetchUserResourceAuthorizations(lastRecord, 100);
+      if (authorizations.isEmpty()) {
+        return;
+      }
+      lastRecord = authorizations.getLast();
+      final Map<String, Map<String, Map<String, List<String>>>> authorizationMap =
+          authorizations.stream()
+              .collect(
+                  Collectors.groupingBy(
+                      UserResourceAuthorization::username,
+                      Collectors.groupingBy(
+                          UserResourceAuthorization::resourceType,
+                          Collectors.groupingBy(
+                              UserResourceAuthorization::permission,
+                              Collectors.mapping(
+                                  UserResourceAuthorization::resourceId, Collectors.toList())))));
+
+      authorizationMap.forEach(
+          (owner, value) -> {
+            value.forEach(
+                (resourceType, permissionAndResources) -> {
+                  final Map<PermissionType, List<String>> permissions =
+                      permissionAndResources.entrySet().stream()
+                          .flatMap(
+                              e ->
+                                  convertPermission(e).stream()
+                                      .map(entry -> new SimpleEntry<>(entry, e.getValue())))
+                          .collect(
+                              Collectors.groupingBy(
+                                  SimpleEntry::getKey,
+                                  Collectors.flatMapping(
+                                      e -> e.getValue().stream(), Collectors.toList())));
+                  final long ownerKey = getOwnerKeyForUsername(owner);
+                  authorizationService.patchAuthorization(
+                      new PatchAuthorizationRequest(
+                          ownerKey,
+                          PermissionAction.ADD,
+                          convertResourceType(resourceType),
+                          permissions));
```

Comment:
🔧 The intention is that this migration app will use the java client to interact with the API, so here instead of using the service classes directly we will use a similar method to the user create via java client you implemented.

I think for now I'm fine if we want to merge the general structure but this will need to change soon.

Reply:
you'r right, I didn't notice that we AuthorizationClient ready.

Reply:
Hmmm I was sure that there was a patch method in the java client, ok something to refactor then, I wonder if a TODO would be good here to not forget NABD though

Reply:
Ah! Its here https://github.com/camunda/camunda/blob/e5c32ef9b87a58085d5197710ec330b96642775d/clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientImpl.java#L627